decisions/0003-package-for-shared-server-features.md

```diff
@@ -0,0 +1,108 @@
+# Package for shared server features
+
+Date: 2022-06-01
+
+Status: proposed
+
+## Context
+
+Currently, users need to install 4 Remix dependencies:
+
+- Dev (`@remix-run/dev`)
+- Server runtime (`@remix-run/{cloudflare,deno,node}`)
+- Adapter (`@remix-run/{arc,cloudflare-pages,cloudflare-workers,express,fly,netlify,vercel}`)
+- Renderer (`@remix-run/react`)
+
+User code currently looks like this:
+```tsx
+// server runtime
+import { json, createFileSessionStorage } from "@remix-run/node";
+// adapter
+import { createRequestHandler } from "@remix-run/express"
+// renderer
+import { useLoaderData } from "@remix-run/react";
+```
+
+### Goals
+
+- make DX and docs simpler: more intuitive, less magic, eliminate some user errors
+- reduce bugs from `remix` internal complexity
+
+### Non-goals
+
+- maintainability of `remix` repo
+- reduce or increase number of packages
+- more portable `app/` directory
```

Comment:
It is currently still the recommended way of changing adapters to just create a new project and copy over the `app` directory though.

https://twitter.com/mjackson/status/1532361967133831174

Not saying we should stick to it, but people will still want to have an easy way to change adapters, so I think we should at least take this into account.

Reply:
Added some prose explaining this^ in the proposal.